cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -697,4 +707,35 @@ private void emitRunTimeMetric(ProgramRunId programRunId, ProgramRunStatus progr
   private AppMetadataStore getAppMetadataStore(StructuredTableContext context) {
     return AppMetadataStore.create(context);
   }
+
+  @Nullable
+  private ProgramRunStatus getProgramRunStatusFromRunRecord(RunRecordDetail record) {
+    if (record == null) {
+      return null;
+    }
+    return record.getStatus();
+  }
+
+  private Integer getPluginCountFromAppMeta(ApplicationMeta applicationMeta) {
```

Comment:
Also, if applicationMeta can be null, annotate it with `@Nullable`

Reply:
ack